dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaServicesTaskTest.java

```diff
@@ -0,0 +1,180 @@
+        argThat(
+            get -> get != null && get.getURI().toString().endsWith("/first-cluster/services"))))
+        .thenReturn(firstResponse);
+
+    CloseableHttpResponse secondResponse = mock(CloseableHttpResponse.class);
+    HttpEntity secondEntity = mock(HttpEntity.class);
+    when(secondResponse.getEntity()).thenReturn(secondEntity);
+    when(secondEntity.getContent())
+        .thenReturn(new ByteArrayInputStream("{\"some\": \"json\"}".getBytes()));
+    when(httpClient.execute(
+        argThat(
+            get -> get != null && get.getURI().toString().endsWith("/second-cluster/services"))))
+        .thenReturn(secondResponse);
+
+    // Act
+    task.doRun(context, sink, handle);
+
+    // Assert
+    Set<URI> requestedUrls = new HashSet<>();
+    verify(httpClient, times(2))
+        .execute(
+            argThat(
+                request -> {
+                  assertEquals(HttpGet.class, request.getClass());
+                  requestedUrls.add(request.getURI());
+                  return true;
+                }));
+    assertEquals(
+        ImmutableSet.of(
+            URI.create("http://localhost/api/clusters/first-cluster/services"),
+            URI.create("http://localhost/api/clusters/second-cluster/services")),
+        requestedUrls);
+
+    // write jsonl. https://jsonlines.org/
+    Set<String> fileLines = new HashSet<>();
+    verify(writer, times(2))
+        .write(
+            (String)
+                argThat(
+                    content -> {
+                      String str = (String) content;
+                      assertFalse(str.contains("\n"));
+                      assertFalse(str.contains("\r"));
+                      fileLines.add(str);
+                      return true;
+                    }));
+    assertEquals(
+        ImmutableSet.of(
+            "{\"some\":\"json\"}",
+            tojsonl(servicesJson)),
+        fileLines);
+
+    verify(firstResponse).close();
+    verify(secondResponse).close();
+    verify(writer).close();
+  }
+
+  @Test
+  public void clustersWereNotInitialized_throwsException() throws Exception {
+    assertNull(handle.getClusters());
+
+    MetadataDumperUsageException exception =
+        assertThrows(MetadataDumperUsageException.class, () -> task.doRun(context, sink, handle));
+
+    assertEquals(
+        "Cloudera clusters must be initialized before services dumping.", exception.getMessage());
+
+    verifyNoWrites();
+  }
+
+
+  private void verifyNoWrites() throws IOException {
+    verify(writer, never()).write(anyChar());
+    verify(writer, never()).write(anyString());
+    verify(writer, never()).write(anyString(), anyInt(), anyInt());
+    verify(writer, never()).write(any(char[].class));
+    verify(writer, never()).write(any(char[].class), anyInt(), anyInt());
+  }
+
+  private String tojsonl(String json) throws Exception {
+    JsonNode jsonNode = new ObjectMapper().readTree(json);
+    return jsonNode.toString();
```

Comment:
can be written in one line:
`return new ObjectMapper().readTree(json).toString();`

Reply:
Done